### ISSUE TITLE
moe: joint grid over hyperparameter and sparsity

### ISSUE THREAD

**Issue by swag2198** [OpenEuroLLM/Taskboard#187]
Do a grid of hyperparameters + sparsity levels to check if optimal hyperparameters are sensitive to sparsity.

Model choice: ~131M active model, topk=2, vary num_experts in [8,16,32,64], total parameters ranging from 322M - 2B
learning rate and batch size grid, say 5 lrs and 5 bszs
Data: 130B tokens (or adjusted to match dense model settings)

Prepared [config](https://github.com/OpenEuroLLM/oellm-autoexp/blob/exp_swagatam/config/experiments/swagatam/test_moe_130M_300BT_filtering.yaml) in autoexp.

**Comment by swag2198** [comment]
See #202 for results.
